frontend/src/components/transactions/ActionRow.tsx

```diff
@@ -40,152 +51,75 @@ export default class extends React.Component<Props, State> {
-            border-bottom: 2px solid #f8f8f8;
-            margin-bottom: 15px;
-            padding-bottom: 8px;
-          }
-
-          .action-compact-row .action-row-img {
-            width: 24px;
-            height: 24px;
-            border: solid 2px #f8f8f8;
-            background-color: #ffffff;
-            border-radius: 50%;
-            margin-right: 8px;
-            text-align: center;
-            line-height: 1.1;
-          }
-
-          .action-sparse-row .action-row-img {
-            margin: 10px 10px 10px 20px;
-            display: inline;
-          }
-
-          .action-sparse-row .action-row-img svg {
-            height: 20px;
-            width: 20px;
-          }
-
-          .action-compact-row .action-row-img svg {
-            height: 12px;
-            width: 12px;
-          }
-
-          .action-sparse-row .action-row-toggler {
-            width: 10px;
-            margin: 10px;
-          }
-
-          .action-row-bottom {
-            border-bottom: solid 2px #f8f8f8;
-          }
-
-          .action-row-title {
-            font-family: BentonSans;
-            font-size: 14px;
-            font-weight: 500;
-            line-height: 1.29;
-            color: #24272a;
-          }
-
-          .action-row-title a {
-            color: #666;
-          }
-
-          .action-row-title a:hover {
-            color: #24272a;
-          }
-
-          .action-row-text {
-            font-family: BentonSans;
-            font-size: 12px;
-            font-weight: 500;
-            line-height: 1.5;
-            color: #999999;
-          }
-
-          .action-row-text a {
-            color: #999999;
-          }
-
-          .action-row-text a:hover {
-            color: #24272a;
-          }
-
-          .action-row-txid {
-            font-family: BentonSans;
-            font-size: 14px;
-            font-weight: 500;
-            line-height: 1.29;
-            color: #0072ce;
-          }
-
-          .action-row-timer {
-            font-family: BentonSans;
-            font-size: 12px;
-            color: #999999;
-            font-weight: 100;
-          }
+    let actionRow;
+    if (detail || batch) {
+      actionRow = (
+        <Row noGutters className={`action-${viewMode}-row mx-0 ${className}`}>
+          <Col xs="auto" className="actions-icon-col">
+            <div className="action-row-img">{ActionIcon && <ActionIcon />}</div>
+          </Col>
+          <Col className="action-row-details">
+            <Row noGutters>
+              <Col md="8" xs="7">
+                <Row noGutters>
+                  <Col className="action-row-title">
+                    <ActionMessage
+                      transaction={transaction}
+                      actionKind={actionKind}
+                      actionArgs={actionArgs}
+                    />
+                  </Col>
+                </Row>
+              </Col>
+            </Row>
+          </Col>
+        </Row>
```

Comment:
There is quite a duplication with the `else` branch. Let's merge it.

Reply:
I tried, but I found it is not worth to merge, because it is not actually a couple duplication, so many different place to split if we really merge them into one. See the <Col> you wrote for the format of the cell. if we just show message and no other message, what I delete for simple actionRow is the simplest.